src/Interpreters/Context.h

```diff
@@ -1180,6 +1181,13 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
     std::shared_ptr<QueryResultCache> getQueryResultCache() const;
     void clearQueryResultCache(const std::optional<String> & tag) const;
 
+#if USE_AVRO
+    void setIcebergMetadataFilesCache(const String & cache_policy, size_t max_size_in_bytes, size_t max_entries, double size_ratio);
+    void updateIcebergMetadataFilesCacheConfiguration(const Poco::Util::AbstractConfiguration & config);
+    std::shared_ptr<IcebergMetadataFilesCache> getIcebergMetadataFilesCache() const;
+    void clearIcebergMetadataFilesCache() const;
```

Comment:
About `clearIcebergMetadataFilesCache`, should we then implement `system drop iceberg metadata cache` (or `system drop data lake metadata cache`?) to put it in use? Similar to `system drop mark cache` (see `InterpreterSystemQuery.cpp`)